pkg/apis/kops/validation/aws.go

```diff
@@ -48,6 +48,13 @@ func awsValidateCluster(c *kops.Cluster) field.ErrorList {
 		allErrs = append(allErrs, awsValidateIAMAuthenticator(field.NewPath("spec", "authentication", "aws"), c.Spec.Authentication.AWS)...)
 	}
 
+	for i, subnet := range c.Spec.Subnets {
```

Comment:
I think we should disallow AdditionalRoutes if the subnet is shared (if the ProviderID field is set)

https://github.com/kubernetes/kops/blob/b2bb322a577ec901de77dbd36e5d6ff12f48d628/pkg/apis/kops/cluster.go#L717-L718

Reply:
I add the check and fail the validation if the `ProviderId` is not empty